icu4c/source/test/intltest/numbertest_api.cpp

```diff
@@ -6085,6 +6086,55 @@ void NumberFormatterApiTest::formatUnitsAliases() {
     }
 }
 
+void NumberFormatterApiTest::formatArbitraryConstant() {
+    IcuTestErrorCode status(*this, "formatArbitraryConstant");
+
+    struct TestCase {
+        const char *unitIdentifier;
+        int32_t inputValue;
+        UNumberUnitWidth width;
+        Locale locale;
+        const UnicodeString expectedOutput;
+    } testCases[]{
+        // {"meter-per-kelvin-second", 2, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getEnglish(),
+        //  "2 meters per second-kelvin"},
+        {"meter-per-100-kelvin-second", 3, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getEnglish(),
+         u"3 meters per 100-second-kelvin"},
+        {"meter-per-1000", 1, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getEnglish(), u"1 meter per 1000"},
+        {"meter-per-1000-second", 1, UNUM_UNIT_WIDTH_FULL_NAME, Locale::getEnglish(),
+         u"1 meter per 1000-second"},
```

Comment:
Didn't you have a bunch of logic for handling plurals in the code above?  Shouldn't this be "1 meter per 1000 seconds" or something like that?  And why is there a hyphen in "1000-second"?  We wouldn't have that in English.

Reply:
unfortunately, for now, it is just a plain number.

There is an issue for that: https://unicode-org.atlassian.net/browse/ICU-23039

Reply:
It looks really weird, but since you have an outstanding ticket to improve this, I'll accept it for now.